modules/@apostrophecms/area/index.js

```diff
@@ -105,6 +105,22 @@ module.exports = {
           self.missingWidgetTypes[name] = true;
         }
       },
+      prepForRender(area, context, fieldName) {
+        if (fieldName.includes('.')) {
+          // If we're passing in a dot path, the area field is the last part.
```

Comment:
This should never happen. It's a holdover from trying to solve arrays the hard way here, right? The forthcoming new implementation of getManagerOf will be able to cope with just context and fieldName, and the context you pass to this is always going to be the immediate parent. Which is the only thing that would work as the rest of this method is currently written.